Formula/node.rb

```diff
@@ -107,12 +89,18 @@ def post_install
       # Dirs must exist first: https://github.com/Homebrew/legacy-homebrew/issues/35969
       mkdir_p HOMEBREW_PREFIX/"share/man/#{man}"
       rm_f Dir[HOMEBREW_PREFIX/"share/man/#{man}/{npm.,npm-,npmrc.,package.json.}*"]
-      ln_sf Dir[libexec/"npm/lib/node_modules/npm/man/#{man}/{npm,package.json}*"], HOMEBREW_PREFIX/"share/man/#{man}"
+      ln_sf Dir[libexec/"npm/man/#{man}/{npm,package.json}*"], HOMEBREW_PREFIX/"share/man/#{man}"
     end
 
     npm_root = node_modules/"npm"
     npmrc = npm_root/"npmrc"
     npmrc.atomic_write("prefix = #{HOMEBREW_PREFIX}\n")
+
+    # This is a minor hack that let's us keep a working npm
+    # in both HOMEBREW_PREFIX & also privately in libexec.
```

Comment:
> The fact I like about the current way we handle install in this PR is, that it brings the way we install npm to HOMEBREW_PREFIX/lib/node_modules/npm more closer to what is upstream supported. 

👍. I do think there are gains to not deviating so hard from upstream so much of the time when it comes to shipping `npm`, not least that when `npm@6` comes out maybe we won't require such a severe rewrite on the brew side of things.

> BTW: If someone has time to try out some of the propsed solutions earlier than me: just go for it.

Thanks. Am trying to throw together something locally. At some point this has to end. Might as well push in that direction.